### PR TITLE
fix(rbs): #MC-164 secure rbs calls

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
Module works when no conf is added

## Checklist tests
choose platform without enableRbs = true config, try to create event

## Issue ticket number and link
#MC-164 https://entsupport.gdapublic.fr/browse/MC-164

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/controllers/controller.ts [hunk 1]
```diff
@@ -716,7 +716,7 @@ export const calendarController = ng.controller('CalendarController',
                  }
             }
 
-            $scope.rbsEmitter.emitBookingInfo(Behaviours.applicationsBehaviours.rbs.eventerRbs.INIT_BOOKING_INFOS, $scope.calendarEvent);
+            if ($scope.ENABLE_RBS) $scope.rbsEmitter.emitBookingInfo(Behaviours.applicationsBehaviours.rbs.eventerRbs.INIT_BOOKING_INFOS, $scope.calendarEvent);
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:719]: secure behaviours calls

File: src/main/resources/public/ts/controllers/controller.ts [hunk 2]
```diff
@@ -735,10 +735,10 @@ export const calendarController = ng.controller('CalendarController',
                 $scope.display.showRecurrencePanel = false;
                 template.close('recurrenceLightbox');
                 $scope.display.showEditEventPanel = true;
-                $scope.rbsEmitter.emitBookingInfo(Behaviours.applicationsBehaviours.rbs.eventerRbs.CAN_EDIT_EVENT, true);
+                if ($scope.ENABLE_RBS) $scope.rbsEmitter.emitBookingInfo(Behaviours.applicationsBehaviours.rbs.eventerRbs.CAN_EDIT_EVENT, true);
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:738]: secure behaviours calls

File: src/main/resources/public/ts/controllers/controller.ts [hunk 3]
```diff
@@ -756,7 +756,7 @@ export const calendarController = ng.controller('CalendarController',
             $scope.display.showViewEventPanel = false;
             $scope.contentToWatch = "";
 
-            $scope.rbsEmitter.emitBookingInfo(Behaviours.applicationsBehaviours.rbs.eventerRbs.CLOSE_BOOKING_INFOS);
+            if ($scope.ENABLE_RBS) $scope.rbsEmitter.emitBookingInfo(Behaviours.applicationsBehaviours.rbs.eventerRbs.CLOSE_BOOKING_INFOS);
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:759]: secure behaviours calls

File: src/main/resources/public/ts/controllers/controller.ts [hunk 4]
```diff
@@ -78,7 +79,7 @@ export const calendarController = ng.controller('CalendarController',
             $scope.buttonAction = ACTIONS;
             $scope.eventSidebar$ = new Subject<void>();
             $scope.ENABLE_RBS = ENABLE_RBS;
-            $scope.rbsEmitter = new RbsEmitter($scope);
+            $scope.rbsEmitter = new RbsEmitter($scope, !!ENABLE_RBS);
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/controller.ts:82]: pourquoi ne pas utiliser $scope.ENABLE_RBS plutôt ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/controllers/controller.ts (hunk 4 above)
Comment (JinDouang) [reply src/main/resources/public/ts/controllers/controller.ts:82]: https://github.com/OPEN-ENT-NG/calendar/blob/51dee1109f993941da84db457e97295169141e2f/src/main/resources/public/ts/controllers/controller.ts#L1420

File: src/main/resources/public/ts/model/rbs/rbs-emitter.model.ts [hunk 5]
```diff
@@ -6,59 +6,70 @@ import {safeApply} from "../Utils";
...
+                    break;
+                case Behaviours.applicationsBehaviours.rbs.eventerRbs.CAN_EDIT_EVENT:
+                    this.scope.$broadcast(event, data);
+                    break;
+                case Behaviours.applicationsBehaviours.rbs.eventerRbs.CLOSE_BOOKING_INFOS:
+                    this.scope.$broadcast(event);
+                    break;
+                case Behaviours.applicationsBehaviours.rbs.eventerRbs.UPDATE_BOOKING_INFOS:
+                    this.scope.$broadcast(event, data);
+                    break;
```
Comment (BastienBenard) [inline src/main/resources/public/ts/model/rbs/rbs-emitter.model.ts:30]: Si tu a plusieurs cas qui font la meme chose tu peux faire:
```
                case Behaviours.applicationsBehaviours.rbs.eventerRbs.INIT_BOOKING_INFOS:
                case Behaviours.applicationsBehaviours.rbs.eventerRbs.CAN_EDIT_EVENT:
                case Behaviours.applicationsBehaviours.rbs.eventerRbs.UPDATE_BOOKING_INFOS:
                    this.scope.$broadcast(event, data);
                    break;
                case Behaviours.applicationsBehaviours.rbs.eventerRbs.CLOSE_BOOKING_INFOS:
                    this.scope.$broadcast(event);
                    break;
```

File: src/main/resources/public/ts/model/rbs/rbs-emitter.model.ts [hunk 6]
```diff
@@ -6,59 +6,70 @@ import {safeApply} from "../Utils";
...
+        }
+
     };
 
     hasAccessToSavedBookings = (): boolean => {
         let rbsSniplet: any = angular.element(document.getElementById("calendar-rbs-booking")).scope();
-        if (rbsSniplet && rbsSniplet.vm) {
+        if (this.ENABLE_RBS && rbsSniplet && rbsSniplet.vm) {
             return rbsSniplet.vm.hasAccessToSavedBookings();
         }
```
Comment (BastienBenard) [inline src/main/resources/public/ts/model/rbs/rbs-emitter.model.ts:67]: il faut returner quelque chose si l'on ne passe pas dans le if

**Review by JinDouang — APPROVED** [review]

File: src/main/resources/public/ts/controllers/controller.ts [hunk 7]
```diff
@@ -1760,7 +1759,8 @@ export const calendarController = ng.controller('CalendarController',
             $scope.noDeleteOptionChosen = (): boolean => {
                 if($scope.calendarEvent.bookings && $scope.calendarEvent.bookings.length > 0) {
                     let bookingDeletionInfo: IScope = angular.element(document.getElementById("booking-deletion-message")).scope();
-                    return bookingDeletionInfo['vm'].canDeleteBooking() && ($scope.calendarEvent.deleteAllBookings == undefined);
+                    return (bookingDeletionInfo && bookingDeletionInfo['vm'] && bookingDeletionInfo['vm'].canDeleteBooking()
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:1762]: eliminate undefined warning

**Review by BastienBenard — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**